src/resolve.rs

```diff
@@ -55,12 +55,18 @@ impl AdmonitionMeta {
 
 /// Make the first letter of `input` upppercase.
 ///
-/// source: https://stackoverflow.com/a/38406885
-fn ucfirst(input: &str) -> String {
-    let mut chars = input.chars();
-    match chars.next() {
-        None => String::new(),
-        Some(f) => f.to_uppercase().collect::<String>() + chars.as_str(),
+/// We special case a few words to make them look nicer (e.g. "tldr" -> "TL;DR" and "faq" -> "FAQ").
+fn as_title_case(input: &str) -> String {
+    match input {
+        "tldr" => "TL;DR".to_owned(),
+        "faq" => "FAQ".to_owned(),
+        "" => String::new(),
+        _ => {
+            // this is necessary to ensure that we deal with UTF-8 correctly rather than byte indices
+            // source: https://stackoverflow.com/a/38406885
+            let mut chars = input.chars();
+            chars.next().unwrap().to_uppercase().collect::<String>() + chars.as_str()
+        }
     }
 }
```

Comment:
Can `ucfirst` be left as-is? I don't see a reason to refactor it in this PR. `as_title_case` can just call it internally.

Also, you're introducing an `unwrap` here (I see that you handle the `""` case explicitly in your new code, but I think it's higher complexity for no actual performance gain).

Reply:
Done in https://github.com/tommilligan/mdbook-admonish/pull/154/commits/fd7b04545f8b232ac3bb0ea5ae40dc0f095eca39